domain/global_vars_cache.go

```diff
@@ -18,6 +18,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/golang/groupcache/singleflight"
```

Comment:
Have you compared this with `golang.org/x/sync/singleflight` ?

Reply:
[golang.org/x/sync/singleflight](https://github.com/golang/sync/blob/master/singleflight/singleflight.go) is a later version, which has more interfaces and fields than [golang/groupcache/singleflight](https://github.com/golang/groupcache/blob/master/singleflight/singleflight.go), better to use the sync package one